users/service.go

```diff
@@ -260,15 +274,15 @@ func (svc service) UpdateClientIdentity(ctx context.Context, token, clientID, id
 func (svc service) GenerateResetToken(ctx context.Context, email, host string) error {
 	client, err := svc.clients.RetrieveByIdentity(ctx, email)
 	if err != nil || client.Credentials.Identity == "" {
-		return repoerr.ErrNotFound
+		return svcerr.ErrNotFound
```

Comment:
Refactor this to something like
```go
client, err := svc.clients.RetrieveByIdentity(ctx, email)
if err != nil {
	return errors.Wrap(svcerr.ErrViewEntity, err)
}
```

This also `ResetSecret` where we check that identity is not empty